### PR TITLE
[CausalLM] SwiGLU layer with forwarding and backpropagation support

### PR DESCRIPTION
**Pull request by niket-agarwal**
- Implement SwiGLU activation function with proper forward and backward passes
- Add  forwarding for training mode
- Add calcDerivative implementation for backpropagation support
- Add comprehensive unit tests with golden data validation
- Generate correct golden data using standard TensorFlow/Keras framework

## Files Modified/Added
- `Applications/CausalLM/layers/swiglu.cpp` 
- `Applications/CausalLM/layers/swiglu.h`
- `test/input_gen/gen_layer_tests.py` 
- `test/unittest/layers/unittest_layers_causallm_swiglu.cpp` 
- `test/unittest/layers/meson.build`

Signed-off-by: Niket Agarwal <niket.a@samsung.com>

### CODE REVIEW

File: test/unittest/layers/meson.build [hunk 1]
```diff
@@ -1,4 +1,4 @@
-layer_common_test_inc = [ include_directories('./', '../../include') ]
+layer_common_test_inc = [ include_directories('./', '../../include', '../../../Applications/CausalLM/layers') ]
```
Comment (EunjuYang) [inline test/unittest/layers/meson.build:1]: It seems not good to add a header in `Application/CausalLM/layers`.
What about moving the swiglu.cpp files to `nntrainer/layers` ?

File: test/unittest/layers/unittest_layers_causallm_swiglu.cpp [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * @date 03 Apr 2026
+ * @brief CausalLM SwiGLU Layer Test
+ * @see	https://github.com/nntrainer/nntrainer
+ * @author Niket Agarwal <niket.a@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#include <tuple>
+#include <gtest/gtest.h>
+#include <layers_common_tests.h>
+#include "../../../Applications/CausalLM/layers/swiglu.h"
```
Comment (EunjuYang) [inline test/unittest/layers/unittest_layers_causallm_swiglu.cpp:15]: Ditto.

File: Applications/CausalLM/layers/swiglu.cpp [hunk 3]
```diff
@@ -32,10 +32,28 @@ float swiglu(float x) { return x / (1 + nntrainer::exp_util(-x)); }
...
+  nntrainer::Tensor &out = context.getOutput(OUT_IDX);
+  nntrainer::Tensor &sig_gate =
+    context.getTensor(tensor_idx[SwiGLUParams::sigmoid_gate]);
+  // sigmoid(gate)
+  gate.apply<float>([](float x) { return 1.0f / (1.0f + nntrainer::exp_util(-x)); },
+                    sig_gate);
+  // swish(gate) = gate * sigmoid(gate)
+  gate.multiply(sig_gate, out);
+  // output = swish(gate) * up
+  out.multiply_i(up);
```
Comment (EunjuYang) [inline Applications/CausalLM/layers/swiglu.cpp:55]: The forwarding operation should be equivalent to incremental_forwarding()'s operation.
Please update this forwarding to act similarly to what incremental_forwarding() does.

**Review by EunjuYang — COMMENTED** [review]
It seems good start to resolve #3836.
However, it seems there are some issues that we have to resolve to make this PR to be merged.

1) Please check whether you update golden dataset for unittest . CI will tries to find the goldendataset used for the new unittest you added. If it cannot find the file, it will fail.

2) It seems not good to add dependency with the files under `Applications/CausalLM/`. How about moving the custom layers under `Applications/CausalLM` into `nntrainer/layers/`?

3) About the forwarding operation, we have to follow the `incremental_forwarding()`'s action. In the `Applicatins/CausalLM`, we focused on optimizing incremental_forwarding opration. So, it would be better the action in incremental_forwarding.

**Comment by niket-agarwal** [comment]
Please refer #3847 for the resolved issues. Closing this one.
